busy-app/busybar-firmware
# Wifi Connection Management

# What's new

- Add a service for managing Wifi connecitons (both main and wireless)
- Add a test application in main firmware

# Caveats

- The `Wifi Setup` does not provide a GUI for selecting a network yet.
- Cannot connect to an access point with WPA3 security and/or in roaming mode.

# How to test

1. Open `applications/main/wifi_setup/wifi_setup.c` and change the following lines according to your network:
```c
#define WIFI_SSID "Your SSID"
#define WIFI_PASS "Your passphrase"
#define WIFI_MODE (WifiSecurityModeWpa2)
```
2. Flash both main and wireless firmware
3. In the main menu, select `Wifi Setup` and watch the serial log output
4. The following procedure should be performed by the app:
   4.1. Initialise the Wifi
   4.2. Scan for access points 3 times (results being printed each time)
   4.3. Connect to the access point with the credentials provided above
   4.4. Wait 5 seconds
   4.5. Disconnect from the access point and deinitialise the Wifi

The procedure should finish without any errors.
